cpp/cmake/Modules/ConfigureArrow.cmake

```diff
@@ -17,23 +35,24 @@ set(ARROW_CMAKE_ARGS " -DARROW_WITH_LZ4=OFF"
                      " -DARROW_JEMALLOC=OFF"
                      " -DARROW_BOOST_VENDORED=OFF"
                      " -DARROW_PYTHON=OFF"
-                     " -DARROW_TENSORFLOW=ON") # enable old ABI for C/C++
-
-configure_file("${CMAKE_SOURCE_DIR}/cmake/Templates/Arrow.CMakeLists.txt.cmake"
-               "${ARROW_ROOT}/CMakeLists.txt")
+                     " -DARROW_HDFS=ON"
+                     " -DARROW_TENSORFLOW=ON" # enable old ABI for C/C++
+                     ) 
 
-file(MAKE_DIRECTORY "${ARROW_ROOT}/build")
-file(MAKE_DIRECTORY "${ARROW_ROOT}/install")
+# Download and unpack arrow at configure time
+configure_file(${CMAKE_SOURCE_DIR}/cmake/Templates/Arrow.CMakeLists.txt.cmake ${ARROW_ROOT}/download/CMakeLists.txt)
 
 execute_process(
-    COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
+    COMMAND ${CMAKE_COMMAND} -F "${CMAKE_GENERATOR}" .
```

Comment:
Why are we using `-F` here?

Reply:
So we wanted to -F force generation. But it seems that it works also with -G option. Understood and corrected.